Aqueous chemical reactions added to PorousFlow

WilkAndy opened the pull request:
Now PorousFlow has basically all the capability of the chemical_reactions module, but it also allows users to couple mechanics, heat flow, etc.  Later @cpgr and I will work together to more closely integrate ChemicalReactions and PorousFlow, but this current addition will allow users to do geochemistry, etc for the time being

Fixes #10847

<!--
INCLUDE THE FOLLOWING IN THE PR DESCRIPTION
- Explain relevant design information for your change.
- Follow the [Coding Standards](http://mooseframework.org/wiki/CodeStandards/).
- Submit or improve [Test Cases](http://mooseframework.org/wiki/MooseTraining/testing/).
- Reference a specific issue, place "refs #<issue>" or "closes #<issue>" (e.g., #closes #1234).
-->

moosebuild commented:
Job [Documentation](https://www.moosebuild.org/job/162482/) on 9d1f280 wanted to post the following:

View the site [here](http://mooseframework.org/docs/PRs/10945/site)

This comment will be updated on new commits.

WilkAndy committed:
Aqueous chemical reactions added to PorousFlow

This commit adds the core functionality to simulate equilibrium as well as kinetic precipitation-dissolution.

Now PorousFlow has basically all the capability of the chemical_reactions module, but it also allows users to couple mechanics, heat flow, etc.  Later @cpgr and I will work together to more closely integrate ChemicalReactions and PorousFlow, but this current addition will allow users to do geochemistry, etc.

Fixes #10847

WilkAndy committed:
Secondary functionality = porosity refactors and enhancements, aux kernel enhancement to record chemical-reaction related things, Dictator enhancements to include chemical-reaction things.

Fixes #10847

WilkAndy committed:
tests of chemical-reactions in PorousFlow
Also small changes to lots of other tests due to refactoring porosity.

Fixes #10847

WilkAndy committed:
doco of chemical-reactions in PorousFlow

Fixes #10847

WilkAndy commented:
Awesome, thanks for the quick merge, @permcody .   I really want to present this chemical-reactions-in-PorousFlow in Switzerland in a couple of weeks, so it's good to know that it's in MOOSE.
